ecl/hql/hqlgram.y

```diff
@@ -6737,6 +6737,9 @@ primexpr1
                         {
                             parser->normalizeExpression($5, type_real, true);
                             OwnedHqlExpr probability = $5.getExpr();
+                            double p = probability->queryValue()->getRealValue();
+                            if (p < 0.0 || p > 1.0)
```

Comment:
I wasn't sure... I thought it may make some automated tools easier to build but I suppose it doesn't make much sense to have 0% and 100% as valid values.